compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -3440,19 +3468,43 @@ object Parsers {
             stats.foreach(checkExtensionMethod(tparams, _))
             ModuleDef(name, Template(makeConstructor(tparams, vparamss), Nil, Nil, self, stats))
           else
-            checkAllGivens(vparamss, "parameter of given instance")
+            def conditionalParents(): List[Tree] =
+              accept(ARROW)
+              if in.token == LPAREN && followingIsParam() then
+                vparamss = vparamss :+ paramClause(vparamss.flatten.length)
```

Comment:
this method of curried function type arguments actually drops arguments:

In the below, the argument of `Show[B]` in `showEither` is not available in implicit scope, and is not a parameter in the rhs
```scala
trait Show[-A] with
  def show(a:A): String

given Show[String] = x => x
given Show[Int] = _.toString

given showEither[A,B]: (sA: Show[A]) => Show[B] => Show[Either[A,B]] = _.fold(a => s"Left(${summon[Show[A]].show(a)})", "Right(Ignored value of B)")

@main def ShowDemo =
  println(summon[Show[Either[Int, String]]].show(Left(-1)))
  println(summon[Show[Either[Int, String]]].show(Right("success message")))
```

Reply:
upon further experimentation, if you give a name to `Show[B]` then it is available in the rhs but still not in implicit scope

Reply:
also, strange that the `Show[Either[A,B]]` example works, but not the `Show[(A,B)]` example:
```scala
trait Show[-A] with
  def show(a:A): String

given Show[String] = x => x
given Show[Int] = _.toString

given showEither[A,B]: (sA: Show[A]) => Show[B] => Show[Either[A,B]] = _.fold(a => s"Left(${summon[Show[A]].show(a)})", _ => "Right(Ignored value of B)")
given [A,B]: (sA: Show[A]) => (sB: Show[B]) => Show[(A,B)] = (a,b) => s"(${sA.show(a)}), ${sB.show(b)})"

@main def ShowDemo =
  println(summon[Show[(Int, String)]].show(0 -> "hello"))
  println(summon[Show[Either[Int, String]]].show(Left(-1)))
  println(summon[Show[Either[Int, String]]].show(Right("success message")))
```

Reply:
I just tried that and it works for me (?), i.e. the example compiles and prints

```
Left(-1)
Right(Ignored value of B)
```

Reply:
Yes, it works but its not possible to use the `... => Show[B] => ...` argument in the rhs, so its just a demo that it is dropped